[Bug]: [Nightly] Search failed among binary collections

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Environment

```markdown
- Milvus version:master(5eb6127)
- Deployment mode(standalone or cluster):standalone
- SDK version(e.g. pymilvus v2.0.0rc2):2.2.0.dev23
- OS(Ubuntu or CentOS): 
- CPU/Memory: 
- GPU: 
- Others:
```


### Current Behavior

Error message:
```
[2022-08-30T15:53:09.176Z] [2022-08-30 15:20:21 - ERROR - ci_test]: search_results_check: limit(topK) searched (8) is not equal with expected (10) (func_check.py:224)
```

### Expected Behavior

PASSED

### Steps To Reproduce

```markdown
nightly:
https://jenkins.milvus.io:18080/blue/organizations/jenkins/Milvus%20Nightly%20CI/detail/master/68/pipeline/123
```


### Milvus Log

1. milvus log: [artifacts-milvus-standalone-nightly-68-pymilvus-e2e-logs.tar.gz](https://jenkins.milvus.io:18080/job/Milvus%20Nightly%20CI/job/master/68/artifact/artifacts-milvus-standalone-nightly-68-pymilvus-e2e-logs.tar.gz)
2. collection name: search_collection_rdaJsCNp
3. FAILED timeline: [2022-08-30T15:20:21.485Z] [gw4] [ 64%] FAILED testcases/test_search.py::TestCollectionSearch::test_search_binary_substructure_flat_index[True-True-False-BIN_FLAT] 
```
[2022-08-30T15:53:09.174Z] =================================== FAILURES ===================================
[2022-08-30T15:53:09.174Z] _ TestCollectionSearch.test_search_binary_substructure_flat_index[True-True-False-BIN_FLAT] _
[2022-08-30T15:53:09.174Z] [gw4] linux -- Python 3.8.13 /usr/local/bin/python3
[2022-08-30T15:53:09.174Z] 
[2022-08-30T15:53:09.174Z] self = <test_search.TestCollectionSearch object at 0x7fbe1819cbb0>
[2022-08-30T15:53:09.174Z] auto_id = True, _async = True, index = 'BIN_FLAT', is_flush = False
[2022-08-30T15:53:09.174Z] 
[2022-08-30T15:53:09.174Z]     @pytest.mark.tags(CaseLabel.L2)
[2022-08-30T15:53:09.174Z]     @pytest.mark.parametrize("index", ["BIN_FLAT"])
[2022-08-30T15:53:09.174Z]     def test_search_binary_substructure_flat_index(self, auto_id, _async, index, is_flush):
[2022-08-30T15:53:09.174Z]         """
[2022-08-30T15:53:09.174Z]         target: search binary_collection, and check the result: distance
[2022-08-30T15:53:09.174Z]         method: compare the return distance value with value computed with SUBSTRUCTURE.
[2022-08-30T15:53:09.174Z]                 (1) The returned limit(topK) are impacted by dimension (dim) of data
[2022-08-30T15:53:09.174Z]                 (2) Searched topK is smaller than set limit when dim is large
[2022-08-30T15:53:09.174Z]                 (3) It does not support "BIN_IVF_FLAT" index
[2022-08-30T15:53:09.174Z]                 (4) Only two values for distance: 0 and 1, 0 means hits, 1 means not
[2022-08-30T15:53:09.174Z]         expected: the return distance equals to the computed value
[2022-08-30T15:53:09.174Z]         """
[2022-08-30T15:53:09.174Z]         # 1. initialize with binary data
[2022-08-30T15:53:09.174Z]         nq = 1
[2022-08-30T15:53:09.174Z]         dim = 8
[2022-08-30T15:53:09.174Z]         collection_w, _, binary_raw_vector, insert_ids, time_stamp \
[2022-08-30T15:53:09.174Z]             = self.init_collection_general(prefix, True, default_nb, is_binary=True, auto_id=auto_id,
[2022-08-30T15:53:09.174Z]                                            dim=dim, is_index=True, is_flush=is_flush)[0:5]
[2022-08-30T15:53:09.174Z]         # 2. create index
[2022-08-30T15:53:09.174Z]         default_index = {"index_type": index, "params": {"nlist": 128}, "metric_type": "SUBSTRUCTURE"}
[2022-08-30T15:53:09.174Z]         collection_w.create_index("binary_vector", default_index)
[2022-08-30T15:53:09.174Z]         collection_w.load()
[2022-08-30T15:53:09.174Z]         # 3. generate search vectors
[2022-08-30T15:53:09.174Z]         _, binary_vectors = cf.gen_binary_vectors(nq, dim)
[2022-08-30T15:53:09.174Z]         # 4. search and compare the distance
[2022-08-30T15:53:09.174Z]         search_params = {"metric_type": "SUBSTRUCTURE", "params": {"nprobe": 10}}
[2022-08-30T15:53:09.174Z] >       res = collection_w.search(binary_vectors[:nq], "binary_vector",
[2022-08-30T15:53:09.174Z]                                   search_params, default_limit, "int64 >= 0",
[2022-08-30T15:53:09.174Z]                                   _async=_async,
[2022-08-30T15:53:09.174Z]                                   travel_timestamp=time_stamp,
[2022-08-30T15:53:09.174Z]                                   check_task=CheckTasks.check_search_results,
[2022-08-30T15:53:09.174Z]                                   check_items={"nq": nq,
[2022-08-30T15:53:09.174Z]                                                "ids": insert_ids,
[2022-08-30T15:53:09.174Z]                                                "limit": default_limit,
[2022-08-30T15:53:09.174Z]                                                "_async": _async})[0]
[2022-08-30T15:53:09.174Z] 
[2022-08-30T15:53:09.174Z] testcases/test_search.py:1781: 
[2022-08-30T15:53:09.174Z] _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
[2022-08-30T15:53:09.174Z] utils/wrapper.py:33: in inner_wrapper
[2022-08-30T15:53:09.174Z]     res, result = func(*args, **kwargs)
[2022-08-30T15:53:09.174Z] base/collection_wrapper.py:162: in search
[2022-08-30T15:53:09.174Z]     check_result = ResponseChecker(res, func_name, check_task, check_items, check,
[2022-08-30T15:53:09.174Z] check/func_check.py:57: in run
[2022-08-30T15:53:09.174Z]     result = self.check_search_results(self.response, self.func_name, self.check_items)
[2022-08-30T15:53:09.174Z] _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
[2022-08-30T15:53:09.174Z] 
[2022-08-30T15:53:09.174Z] search_res = <pymilvus.orm.search.SearchResult object at 0x7fbe19360190>
[2022-08-30T15:53:09.174Z] func_name = 'search'
[2022-08-30T15:53:09.174Z] check_items = {'_async': True, 'ids': [435649985361230814, 435649985361230815, 435649985361230816, 435649985361230817, 435649985361230818, 435649985361230819, ...], 'limit': 10, 'nq': 1}
[2022-08-30T15:53:09.174Z] 
[2022-08-30T15:53:09.174Z]     @staticmethod
[2022-08-30T15:53:09.174Z]     def check_search_results(search_res, func_name, check_items):
[2022-08-30T15:53:09.174Z]         """
[2022-08-30T15:53:09.174Z]         target: check the search results
[2022-08-30T15:53:09.174Z]         method: 1. check the query number
[2022-08-30T15:53:09.174Z]                 2. check the limit(topK) and ids
[2022-08-30T15:53:09.174Z]                 3. check the distance
[2022-08-30T15:53:09.174Z]         expected: check the search is ok
[2022-08-30T15:53:09.174Z]         """
[2022-08-30T15:53:09.174Z]         log.info("search_results_check: checking the searching results")
[2022-08-30T15:53:09.174Z]         if func_name != 'search':
[2022-08-30T15:53:09.174Z]             log.warning("The function name is {} rather than {}".format(func_name, "search"))
[2022-08-30T15:53:09.175Z]         if len(check_items) == 0:
[2022-08-30T15:53:09.175Z]             raise Exception("No expect values found in the check task")
[2022-08-30T15:53:09.175Z]         if check_items.get("_async", None):
[2022-08-30T15:53:09.175Z]             if check_items["_async"]:
[2022-08-30T15:53:09.175Z]                 search_res.done()
[2022-08-30T15:53:09.175Z]                 search_res = search_res.result()
[2022-08-30T15:53:09.175Z]         if len(search_res) != check_items["nq"]:
[2022-08-30T15:53:09.175Z]             log.error("search_results_check: Numbers of query searched (%d) "
[2022-08-30T15:53:09.175Z]                       "is not equal with expected (%d)"
[2022-08-30T15:53:09.175Z]                       % (len(search_res), check_items["nq"]))
[2022-08-30T15:53:09.175Z]             assert len(search_res) == check_items["nq"]
[2022-08-30T15:53:09.175Z]         else:
[2022-08-30T15:53:09.175Z]             log.info("search_results_check: Numbers of query searched is correct")
[2022-08-30T15:53:09.175Z]         for hits in search_res:
[2022-08-30T15:53:09.175Z]             if (len(hits) != check_items["limit"]) \
[2022-08-30T15:53:09.175Z]                     or (len(hits.ids) != check_items["limit"]):
[2022-08-30T15:53:09.175Z]                 log.error("search_results_check: limit(topK) searched (%d) "
[2022-08-30T15:53:09.175Z]                           "is not equal with expected (%d)"
[2022-08-30T15:53:09.175Z]                           % (len(hits), check_items["limit"]))
[2022-08-30T15:53:09.175Z] >               assert len(hits) == check_items["limit"]
[2022-08-30T15:53:09.175Z] E               AssertionError
[2022-08-30T15:53:09.175Z] 
[2022-08-30T15:53:09.175Z] check/func_check.py:227: AssertionError
```



### Anything else?

_No response_

--- a/tests/python_client/testcases/test_search.py
+++ b/tests/python_client/testcases/test_search.py
@@ -1778,16 +1778,12 @@ class TestCollectionSearch(TestcaseBase):
         res = collection_w.search(binary_vectors[:nq], "binary_vector",
                                   search_params, default_limit, "int64 >= 0",
                                   _async=_async,
-                                  travel_timestamp=time_stamp,
-                                  check_task=CheckTasks.check_search_results,
-                                  check_items={"nq": nq,
-                                               "ids": insert_ids,
-                                               "limit": default_limit,
-                                               "_async": _async})[0]
+                                  travel_timestamp=time_stamp)[0]
         if _async:
             res.done()
             res = res.result()
         assert res[0].distances[0] == 0.0
+        assert len(res) <= default_limit
 
     @pytest.mark.tags(CaseLabel.L2)
     @pytest.mark.parametrize("index", ["BIN_FLAT"])

